language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java

```diff
@@ -291,14 +291,20 @@ public ContextTypeResolver(PositionedOperationContext context) {
             return Optional.empty();
         }
         Optional<TypeSymbol> typeRefSymbol = semanticModel.get().typeOf(errorConstructorExpressionNode);
-        if (typeRefSymbol.isEmpty() || typeRefSymbol.get().typeKind() != TypeDescKind.TYPE_REFERENCE) {
+        if (typeRefSymbol.isEmpty()) {
             return Optional.empty();
         }
-        TypeSymbol typeSymbol = ((TypeReferenceTypeSymbol) typeRefSymbol.get()).typeDescriptor();
-        if (typeSymbol.typeKind() != TypeDescKind.ERROR) {
-            return Optional.empty();
+        if (typeRefSymbol.get().typeKind() == TypeDescKind.ERROR) {
+            return Optional.ofNullable(((ErrorTypeSymbol) typeRefSymbol.get()).detailTypeDescriptor());
+        }
+        if (typeRefSymbol.get().typeKind() == TypeDescKind.TYPE_REFERENCE) {
+            TypeSymbol typeSymbol = ((TypeReferenceTypeSymbol) typeRefSymbol.get()).typeDescriptor();
+            if (typeSymbol.typeKind() != TypeDescKind.ERROR) {
+                return Optional.empty();
+            }
+            return Optional.ofNullable(CommonUtil.getRawType(((ErrorTypeSymbol) typeSymbol).detailTypeDescriptor()));
```

Comment:
Shall we refactor this part? We have the same code segment repeat in a slightly different manner.  Also, add a comment as well, mentioning the two use-cases for clarity.